src/main/java/org/broadinstitute/hellbender/tools/sv/SVCallRecordUtils.java

```diff
@@ -0,0 +1,467 @@
+                } else {
+                    throw new IllegalStateException("Genotype for sample " + g.getSampleName() + " at " + variant.getContig() + ":" + variant.getStart() + " had no CN attribute and will be dropped.");
+                }
+            } else {
+                throw new IllegalArgumentException("Segment VCF schema expects <DEL>, <DUP>, and no-call allele, but found "
+                        + g.getAllele(0) + " at " + variant.getContig() + ":" + variant.getStart());
+            }
+        }
+        final boolean startStrand = isDel ? true : false;
+        final boolean endStrand = isDel ? false : true;
+        final StructuralVariantType type;
+        final List<Allele> alleles;
+        if (!variant.getReference().equals(Allele.REF_N) && variant.getAlternateAlleles().contains(GATKSVVCFConstants.DUP_ALLELE)
+                && variant.getAlternateAlleles().contains(GATKSVVCFConstants.DEL_ALLELE)) {
+            type = StructuralVariantType.CNV;
+            alleles = variant.getAlleles();
+        } else if (isDel) {
+            type = StructuralVariantType.DEL;
+            alleles = Lists.newArrayList(variant.getReference(), GATKSVVCFConstants.DEL_ALLELE);
+        } else {
+            type = StructuralVariantType.DUP;
+            alleles = Lists.newArrayList(variant.getReference(), GATKSVVCFConstants.DUP_ALLELE);
+        }
+
+        final String id = variant.getID();
+        final String startContig = variant.getContig();
+        final int start = variant.getStart();
+        final int end = variant.getEnd();
+        final int length = end - start;
+        final List<String> algorithms = Collections.singletonList(GATKSVVCFConstants.DEPTH_ALGORITHM);
+        return new SVCallRecord(id, startContig, start, startStrand, startContig, end, endStrand, type, length, algorithms, alleles, variant.getGenotypes(), variant.getAttributes());
+    }
+
+    /**
+     * @param g
+     * @return true if this is a call on a missing contig
+     */
+    private static boolean isNullCall(final Genotype g) {
+        return g.hasExtendedAttribute(GATKSVVCFConstants.COPY_NUMBER_FORMAT)
+                && VariantContextGetters.getAttributeAsInt(g, GATKSVVCFConstants.COPY_NUMBER_FORMAT, 0) == 0
+                && g.isNoCall();
+
+    }
+
+    public static boolean containsAltAllele(final Genotype g) {
+        return g.getAlleles().stream().anyMatch(SVCallRecordUtils::isAltAllele);
+    }
+
+    public static boolean isAltAllele(final Allele allele) {
+        return allele != null && !allele.isNoCall() && !allele.isReference();
+    }
+
+    // TODO this is sort of hacky but the Allele compareTo() method doesn't give stable ordering
+    public static List<Allele> sortAlleles(final Collection<Allele> alleles) {
+        return alleles.stream().sorted(Comparator.nullsFirst(Comparator.comparing(Allele::getDisplayString))).collect(Collectors.toList());
```

Comment:
It's unlikely to have much impact, except for multi-allelic CNVs where it will slightly prefer DEL over DUP when collapsing genotypes. This part of the code may or may not need some shaking out later after testing with some real data.